js/plugins/express/src/index.ts

```diff
@@ -14,88 +14,62 @@
  * limitations under the License.
  */
 
+import { RequestData, getCallableJSON, getHttpStatus } from '@genkit-ai/core';
 import * as bodyParser from 'body-parser';
 import cors, { CorsOptions } from 'cors';
-import express, { RequestHandler } from 'express';
-import { Action, Flow, runWithStreamingCallback, z } from 'genkit';
+import express from 'express';
+import {
+  Action,
+  ActionContext,
+  ContextProvider,
+  Flow,
+  runWithStreamingCallback,
+  z,
+} from 'genkit';
 import { logger } from 'genkit/logging';
 import { Server } from 'http';
-import { getErrorMessage, getErrorStack } from './utils';
 
 const streamDelimiter = '\n\n';
 
-/**
- * Auth policy context is an object passed to the auth policy providing details necessary for auth.
- */
-export interface AuthPolicyContext<
-  I extends z.ZodTypeAny = z.ZodTypeAny,
-  O extends z.ZodTypeAny = z.ZodTypeAny,
-  S extends z.ZodTypeAny = z.ZodTypeAny,
-> {
-  action?: Action<I, O, S>;
-  input: z.infer<I>;
-  auth?: Record<string, any>;
-  request: RequestWithAuth;
-}
-
-/**
- * Flow Auth policy. Consumes the authorization context of the flow and
- * performs checks before the flow runs. If this throws, the flow will not
- * be executed.
- */
-export interface AuthPolicy<
-  I extends z.ZodTypeAny = z.ZodTypeAny,
-  O extends z.ZodTypeAny = z.ZodTypeAny,
-  S extends z.ZodTypeAny = z.ZodTypeAny,
-> {
-  (ctx: AuthPolicyContext<I, O, S>): void | Promise<void>;
-}
-
-/**
- * For express-based flows, req.auth should contain the value to bepassed into
- * the flow context.
- */
-export interface RequestWithAuth extends express.Request {
-  auth?: Record<string, any>;
-}
-
 /**
  * Exposes provided flow or an action as express handler.
  */
 export function expressHandler<
+  C extends ActionContext = ActionContext,
   I extends z.ZodTypeAny = z.ZodTypeAny,
   O extends z.ZodTypeAny = z.ZodTypeAny,
   S extends z.ZodTypeAny = z.ZodTypeAny,
 >(
   action: Action<I, O, S>,
   opts?: {
-    authPolicy?: AuthPolicy<I, O, S>;
+    context?: ContextProvider<C, I>;
```

Comment:
I feel just `context` feels like it takes the context object... perhaps `contextProvider`? I don't know... not feeling too strongly.